src/NuGet.Clients/NuGet.CommandLine/NuGetCommand.resx

```diff
@@ -318,7 +318,7 @@ nuget pack foo.nuspec -Version 2.1.0</value>
     <value>Pushes a package to the server and publishes it.</value>
   </data>
   <data name="PushCommandSourceDescription" xml:space="preserve">
-    <value>Specifies the server URL. If not specified, nuget.org is used unless DefaultPushSource config value is set in the NuGet config file.</value>
+    <value>Specifies the server URL. NuGet identifies a UNC or local folder source and simply copies the file there instead of pushing it using HTTP. Also, starting with NuGet 3.4.2, this is a mandatory parameter unless the NuGet.Config file specifies a DefaultPushSource value.</value>
```

Comment:
Feels this jumps very quickly to talking about local sources and shares. 
I don't think we need to mention the version in a help command. 

How about: 

> Specifies the server URL. If not specified, the DefaultPushSource value from the NuGet.Config will be used. If NuGet identifies a UNC or local folder source and simply copies the file there instead of pushing it using HTTP. Mandatory if the NuGet.Config does not specify a default value.

@zivkan @anangaur thoughts?

Reply:
Oh sorry, it seems that I pushed an old version here when I changed the Designer file. I did remove the version reference in an earlier adjustment. Will update the PR in a moment.

Reply:
Do you mind flipping the order of the statements? 
What happens if not specified is higher priority than the local push explanation.

Reply:
Sorry for the delay on that, I’ve just updated the PR to reorder that. Do you also still want to split up the not-specified behavior from the mandatory explanation (like in your suggestion above)?

Reply:
I'd like to get another pair of eyes from @anangaur and/or @rrelyea